dll/win32/shdocvw/CFavBand.cpp

```diff
@@ -0,0 +1,648 @@
+                                    ::GetModuleHandleW(NULL), NULL);
+    ATLASSERT(hwndTV);
+    if (!hwndTV)
+        return FALSE;
+
+    m_hwndTreeView.Attach(hwndTV);
+    TreeView_SetImageList(m_hwndTreeView, m_hTreeViewImageList, TVSIL_NORMAL);
+
+    return TRUE;
+}
+
+// *** message handlers ***
+
+LRESULT CFavBand::OnCreate(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    if (!CreateToolbar() || !CreateTreeView())
+        return -1;
+
+    return 0;
+}
+
+LRESULT CFavBand::OnDestroy(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    m_hwndTreeView.DestroyWindow();
+    m_hwndToolbar.DestroyWindow();
+    return 0;
+}
+
+LRESULT CFavBand::OnSize(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    if (!m_hwndTreeView)
+        return 0;
+
+    RECT rc;
+    GetClientRect(&rc);
+    LONG cx = rc.right, cy = rc.bottom;
+
+    RECT rcTB;
+    m_hwndToolbar.SendMessage(TB_AUTOSIZE, 0, 0);
+    m_hwndToolbar.GetWindowRect(&rcTB);
+
+    LONG cyTB = rcTB.bottom - rcTB.top;
+    m_hwndTreeView.MoveWindow(0, cyTB, cx, cy - cyTB);
+
+    return 0;
+}
+
+LRESULT CFavBand::OnSetFocus(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    m_bFocused = TRUE;
+    IUnknown_OnFocusChangeIS(m_pSite, reinterpret_cast<IUnknown*>(this), TRUE);
+    return 0;
+}
+
+LRESULT CFavBand::OnKillFocus(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    IUnknown_OnFocusChangeIS(m_pSite, reinterpret_cast<IUnknown*>(this), FALSE);
+    m_bFocused = FALSE;
+    return 0;
+}
+
+LRESULT CFavBand::OnCommand(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    switch (LOWORD(wParam))
+    {
+        case ID_ADD:
+        {
+            UNIMPLEMENTED;
+            MessageBox(L"Not implemented yet", NULL, MB_ICONERROR);
+            break;
+        }
+        case ID_ORGANIZE:
+        {
+            SHELLEXECUTEINFOW sei = { sizeof(sei), SEE_MASK_IDLIST };
+            sei.hwnd = m_hWnd;
+            sei.nShow = SW_SHOWNORMAL;
+            sei.lpIDList = m_pidlFav;
+            ::ShellExecuteExW(&sei);
+            break;
+        }
+    }
+    return 0;
+}
+
+// *** IOleWindow ***
+
+STDMETHODIMP CFavBand::GetWindow(HWND *lphwnd)
+{
+    if (!lphwnd)
+        return E_INVALIDARG;
+    *lphwnd = m_hWnd;
+    return S_OK;
+}
+
+STDMETHODIMP CFavBand::ContextSensitiveHelp(BOOL fEnterMode)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+// *** IDockingWindow ***
+
+STDMETHODIMP CFavBand::CloseDW(DWORD dwReserved)
+{
+    // We do nothing, we don't have anything to save yet
+    TRACE("CloseDW called\n");
+    return S_OK;
+}
+
+STDMETHODIMP CFavBand::ResizeBorderDW(const RECT *prcBorder, IUnknown *punkToolbarSite, BOOL fReserved)
+{
+    /* Must return E_NOTIMPL according to MSDN */
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CFavBand::ShowDW(BOOL fShow)
+{
+    m_fVisible = fShow;
+    ShowWindow(fShow ? SW_SHOWNORMAL : SW_HIDE);
```

Comment:
Just SW_SHOW? Show normal is for top level windows?

Reply:
Fixed.